source/service.go

```diff
@@ -129,6 +130,23 @@ func (sc *serviceSource) endpointsFromTemplate(svc *v1.Service) ([]*endpoint.End
 	return endpoints, nil
 }
 
+func getTTLFromAnnotations(annotations map[string]string) endpoint.TTL {
```

Comment:
This function belongs to ingress and service, so this should be factored out into another file that is for both.
Maybe there is already a file where it could fit.